validator/src/main/java/com/google/udmi/util/git/GitRepositoryInterface.java

```diff
@@ -0,0 +1,70 @@
+package com.google.udmi.util.git;
+
+import java.io.IOException;
+import java.util.List;
+import org.eclipse.jgit.api.Status;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.FetchResult;
+import org.eclipse.jgit.transport.PushResult;
+
+/**
+ * Interface for a Git Repository.
+ */
+public interface GitRepositoryInterface extends AutoCloseable {
```

Comment:
At this point, I think it would be prudent to introduce one additional layer of abstraction which is just `sourceRepository` that doesn't expose _everything_ git has to offer, and doesn't have any explicit ties to git -- e.g. conceptualize it just as being for any repository.  Should be fairly easy to do -- just have the new interface only implement the actual functions that are used by BAMBI.   It's generally easier to add things later than it is to take things away (which might break something).